### ISSUE TITLE
Exclude the load balancing loss of padding tokens in Mixtral-8x7B

### ISSUE THREAD

**Issue by khaimt** [huggingface/transformers#28505]
### Feature request

The auxiliary loss in Mixtral-MoE shouldn't **include the loss from padding tokens**. 

### Motivation

I think it is better to change the function 
[load_balancing_loss_func](https://github.com/huggingface/transformers/blob/main/src/transformers/models/mixtral/modeling_mixtral.py#L77) by adding an additional parameter: `attention_mask` and change the implementation inside to remove the loss from padding tokens

### Your contribution

I would be happy to review the PR implemeting this feature !

**Comment by amyeroberts** [comment]
cc @ArthurZucker

**Comment by AvinashDwivedi** [comment]
Hi @ArthurZucker,

I hope this message finds you well. I noticed the reported issue and wanted to express my willingness to contribute. Please feel free to assign the task to me, and I'll do my best to address it promptly. Looking forward to collaborating on this!

Best regards,
Avinash Kumar Dubey

**Comment by avisinghal6** [comment]
I would like to work on this issue, i will go through the linked file today and ask any questions i have.

**Comment by nakranivaibhav** [comment]
I was looking at the code.
Below is what the model outputs
`return MoeModelOutputWithPast(
            last_hidden_state=hidden_states,
            past_key_values=next_cache,
            hidden_states=all_hidden_states,
            attentions=all_self_attns,
            router_logits=all_router_logits,
        )`
        
The attention from the model output can be passed during load_balancing_loss_func, and the function can be changed appropriately to handle the pad tokens.
Am I right in my understanding? @ArthurZucker

**Comment by khaimt** [comment]
> feel free to open a PR for this! Otherwise will mark it as a good second issue 🤗

Hi @ArthurZucker, 
I have just raised a PR: 
https://github.com/huggingface/transformers/pull/28513
